src/jvmMain/kotlin/io/beatmaps/api/collaboration.kt

```diff
@@ -152,17 +154,20 @@ fun Route.collaborationRoute() {
                                 recipients
                             )
                         }
-                        true
+                        Pair(true, map)
                     } else {
-                        false
+                        Pair(false, null)
                     }
                 } else {
-                    Collaboration.deleteWhere {
+                    val success = Collaboration.deleteWhere {
                         id eq req.collaborationId and (collaboratorId eq sess.userId)
                     } > 0
+
+                    Pair(success, null)
```

Comment:
But doesn't this mean that the message won't be sent? map == null
Given you can't get a map object here it might make more sense for the second result to also be a boolean and return `map != null` above instead of `map`

You'll also want to use `req.mapId` below instead of using the map object.

Reply:
Exactly! This is pretty much intended.

This is the case where the collaboration is rejected, hence no change. I find it bit awkward too though, best I could think of define the returned second parameter to be null if not successful accepted.

I did try to use `req.mapId`, but I quickly realized it doesn't exist, because the collaboration request object is denied or accepted and only that ID is passed.